### ISSUE TITLE
 appimaged-x86_64.AppImage missing libarchive.so.13

### ISSUE THREAD

**Issue by ShalokShalom** [AppImage/AppImageKit#343]
when i run chmod +x on `appimaged-x86_64.AppImage` and execute it:

appimaged: error while loading shared libraries: libarchive.so.13: cannot open shared object file: No such file or directory

and when i run `appimagetool-x86_64.AppImage` 

WARNING: appstreamcli is missing, please install it if you want to use AppStream metadata
WARNING: appstream-util is missing, please install it if you want to use AppStream metadata
SOURCE is missing

That counts for both, continuous build and official release. 

The [Etcher](etcher.io) AppImage runs even without chmod, i simply click on the file and i starts there.
When I do the same with appimaged, launches K3B, since it is meant to handle image files.

**Comment by probonopd** [comment]
Thanks @ShalokShalom which distribution and version do you use?

`appimagetool` needs a `SOURCE` argument:

```
Usage:
  appimagetool [OPTION...] SOURCE [DESTINATION] - Generate, extract, and inspect AppImages

Help Options:
  -h, --help                  Show help options

Application Options:
  -l, --list                  List files in SOURCE AppImage
  -u, --updateinformation     Embed update information STRING; if zsyncmake is installed, generate zsync file
  --bintray-user              Bintray user name
  --bintray-repo              Bintray repository
  --version                   Show version number
  -v, --verbose               Produce verbose output
  -s, --sign                  Sign with gpg2
  -n, --no-appstream          Do not check AppStream metadata
```

**Comment by ShalokShalom** [comment]
KaOS

**Comment by darealshinji** [comment]
What happened to the GUI wizard?

**Comment by ShalokShalom** [comment]
GUI wizard?

**Comment by ShalokShalom** [comment]
https://github.com/probonopd/AppImageKit/issues/342

**Comment by probonopd** [comment]
The "GUI wizard" AppImageAssistant will not receive updates anymore since has been replaced by `appimagetool`.

**Comment by probonopd** [comment]
Might be caused bhttps://github.com/libarchive/libarchive/issues/817

**Comment by darealshinji** [comment]
Is there a reason to not link libarchive statically or to load it using `dlopen(3)`? In the latter case you could check whether `libarchive.so.13` or `libarchive.so.3` is present.

**Comment by ShalokShalom** [comment]
@probonopd I cant see any connection - how to test? (Etcher work, as written)

**Comment by probonopd** [comment]
libarchive.so.13 is needed by appimaged (the application inside the appimaged.AppImage). libarchive is installed on all Linux desktop distributions, but they all can't agree on a version number. Some call it "13", others call it other numbers. I think this is what may be causing trouble here. Even though in reality they are all version 3 and would all work, if they were named correctly.

A workaround might be to bundle yet another copy of libarchive in the appimaged.AppImage.

**Comment by ShalokShalom** [comment]
Whats here on my computer: 

core/libarchive 3.2.2-1 /usr/lib/libarchive.a
core/libarchive 3.2.2-1 /usr/lib/libarchive.so
core/libarchive 3.2.2-1 /usr/lib/libarchive.so.15
core/libarchive 3.2.2-1 /usr/lib/pkgconfig/libarchive.pc
core/libarchive 3.2.2-1 /usr/share/licenses/libarchive/COPYING                                                                                                         
core/libarchive 3.2.2-1 /usr/share/man/man3/libarchive.3.gz                                                                                                            
core/libarchive 3.2.2-1 /usr/share/man/man3/libarchive_internals.3.gz                                                                                                  
core/libarchive 3.2.2-1 /usr/share/man/man5/libarchive-formats.5.gz

**Comment by probonopd** [comment]
So your distribution thinks it needs to name it "15" but appimagetool thinks it needs "13". Although most likely both are version 3 in fact.

We still need to figure out how to handle this best.

**Comment by ShalokShalom** [comment]
hnn, this seems to happen on compilation, since there is no 15 at all in the source files.. https://github.com/KaOSx/core/tree/master/libarchive

**Comment by darealshinji** [comment]
I'm getting soversion 15 too. Must have to do with this part of CMakeLists.txt:
``` cmake
# INTERFACE_VERSION increments with every release
# libarchive 2.7 == interface version 9 = 2 + 7
# libarchive 2.8 == interface version 10 = 2 + 8
# libarchive 2.9 == interface version 11 = 2 + 9
# libarchive 3.0 == interface version 12
# libarchive 3.1 == interface version 13
math(EXPR INTERFACE_VERSION  "13 + ${_minor}")
```
Should actually be 16 then. Version 3.2 + minor 2 = 14 + 2.

**Comment by probonopd** [comment]
Assuming that all 3.x versions are backward API compatible within the 3.x series (as suggested by [semver](http://semver.org/)), shouldn't it be named *.so.3 and be done with it?

**Comment by ShalokShalom** [comment]
Did you ask similar projects how they handle that stuff?

reddit.com/r/linux/comments/4ohvur/nix_vs_snap_vs_flatpak_what_are_the_differences/

**Comment by probonopd** [comment]
It would be better to remove `libarchive`, see https://github.com/AppImage/AppImageKit/issues/353

**Comment by TheAssassin** [comment]
Closed in #355.
